src/content/drafts/cesium-eslint-case-study.md

```diff
@@ -0,0 +1,41 @@
+
+
+## How we use ESLint
+
+The earlier in the development process a developer is able to catch and fix an issue, the less expensive it is to fix it. This “shifting left” strategy is a key priority for us and that is why we incorporate linting into the developer workflow so that developers are informed of common potential issues in their code as they write it or before they commit it. First, Visual Studio Code is the IDE of choice at Cesium, so we include the official ESLint VS Code extension in .vscode/extensions.json so that users are prompted to install the extension when they clone the repository and open it in VS Code. 
+
+![VSCode ESLint extension](/assets/images/blog/2022/eslint-vscode.png)
+
+Second, we use the [husky](https://github.com/typicode/husky) package to install pre-commit hooks to ensure that the lint command is run every time code is committed to the repository. We also run the lint script as part of our CI step. These tools are replicated for all our repositories to ensure a consistent and efficient experience in development and in review. Maintaining consistency in the code and the tools reduces the burden on developers working on different projects and makes it easier to onramp developers on new projects.
+
+At Cesium, we maintain two [open source shareable ESLint configurations](https://github.com/CesiumGS/eslint-config-cesium): one for Node projects and another for browser environments. Often, both configurations are used in the same repository as there is often server/script code to accompany the client/web code. Our configurations rely heavily on the default ESLint recommended ruleset. Since we use Prettier for code formatting in all our projects, we also add the Prettier ruleset to ensure that there are no conflicts between the linter and the formatter. Projects often also end up extending the base source configuration to create a testing configuration for the Jasmine environment.
+
+Up until 2022, CesiumJS had been locked to ECMAScript 2009, primarily due to browser constraints imposed by supporting Internet Explorer. We used [eslint-plugin-es](https://eslint-plugin-es.mysticatea.dev/) to restrict or allow specific language features. Once we dropped support for Internet Explorer, we were free to upgrade to a more modern version of ECMAScript. We set our version to ECMAScript 2020, and through ESLint the process was mostly automated thanks to the fix command. There were a lot of benefits that were instantly apparent to us; to name a couple of examples, GLSL shader strings were much easier to read because of the string template syntax and let and const made the code a lot more readable through clearer scoping and enforcing immutable references where needed.
```

Comment:
```suggestion
Prior to 2022, CesiumJS had been locked to ECMAScript 2009, primarily due to browser constraints imposed by supporting Internet Explorer. We used [eslint-plugin-es](https://eslint-plugin-es.mysticatea.dev/) to restrict or allow specific language features. Once we dropped support for Internet Explorer, we were free to upgrade to a more modern version of ECMAScript. We set our version to ECMAScript 2020, and through ESLint the process was mostly automated thanks to the fix command. There were a lot of benefits that were instantly apparent to us; to name a couple of examples, GLSL shader strings were much easier to read because of the string template syntax and let and const made the code a lot more readable through clearer scoping and enforcing immutable references where needed.
```